### DISCUSSION TITLE
Does the IPv6 configuration of the main router affect OTBR?

### DISCUSSION THREAD

**Discussion by xiaojiang2017** [openthread/openthread#11084] [Q&A]
**Describe the bug** A clear and concise description of what the bug is.
Hi Experts,
The topology diagram is as follows,
```
       MainRouter
      /        \
  PC          OTBR
                  \
                    ED
```
The OTBR uses SLAAC by default and does not enable DHCP6 relative.
According to our customer’s feedback, if the MainRouter uses stateful, the PC can ping to ED, but if the MainRouter uses stateless, the PC can not ping to ED.

1. Git commit id **simplicity_sdk 2024.6.1**
2. IEEE 802.15.4 hardware platform **EFR32MG21F768**

**Comment by jwhui** [comment]
Devices on both the Thread and Infrastructure networks need to have routable addresses. A routable address is one that has non-link-local scope and can be either unique-local or global prefixes.

With IPv6, SLAAC prefixes on a Wi-Fi / Ethernet network are configured using IPv6 Router Advertisements that contain a Prefix Information Option. Hosts (including the PC in your picture) should autoconfigure an address upon receiving the IPv6 RAs.

In environments where no router is advertising a routable IPv6 prefix, the OTBR should advertise its own IPv6 RAs and the hosts (including the PC in your picture) should configure routable IPv6 addresses using the RAs sent by the OTBR.

To debug your issue further, check to make sure that the PC has configured routable IPv6 addresses and routes to the IPv6 prefix assigned to the Thread network.

**Reply by xiaojiang2017** [reply]
Maybe I know the root cause. The main router allocates the global IPv6 address prefix and uses a stateless address. However, OTBR uses the automatically generated OMR address prefix and does not use the global IPv6 address prefix allocated by the main router, which causes this problem. 

What I want to know is if OTBR implementation can automatically use the global IPv6 prefix assigned by the main router to generate IPv6 addresses. Or does the host system need to call the OT API to set it?

**Reply by jwhui** [reply]
Yes, OTBR tracks the suitable on-link prefixes on the infrastructure network. A suitable on-link prefix must have the on-link flag set and either the auto-configuration flag or DHCPv6 preferred flag set.

https://github.com/openthread/openthread/blob/151444f7dd808aa8fb1431804a6f5649e0569b17/src/core/border_router/routing_manager.cpp#L675

Note that Off-Mesh Routable (OMR) prefix is only assigned to the Thread network.

As mentioned above, I would check whether the PC has a route configured for the Thread network's prefix and an address that is configured out of a suitable on-link prefix.

**Reply by xiaojiang2017** [reply]
Does OTBR have any restrictions about on-link prefix length? Only support `::/64`?

**Reply by jwhui** [reply]
Yes, the on-link prefix must have length 64. See implementation here: https://github.com/openthread/openthread/blob/151444f7dd808aa8fb1431804a6f5649e0569b17/src/core/border_router/routing_manager.cpp#L685
